detector/other/FieldMapBrBz.cpp

```diff
@@ -150,44 +157,156 @@ void FieldMapBrBz::fillFieldMapFromTree(const std::string& filename, const std::
+      BrhoVar = strs[i];
+    }
+    else if(strs[i].find(std::string("Bz")) != std::string::npos) {
+      BzVar   = strs[i];
+    }
+    else if(strs[i].find(std::string("rho")) != std::string::npos || strs[i].find(std::string("Rho")) != std::string::npos) {
+      rhoVar  = strs[i];
+    }
+    else if(strs[i].find(std::string("z")) != std::string::npos || strs[i].find(std::string("Z")) != std::string::npos) {
+      zVar    = strs[i];
+    }
+    else {
+      NtupleName = strs[i];
+    }
+  }
+
+  if(NtupleName == std::string("")) {
+    std::stringstream error;
+    error << "FieldMapBrBz[ERROR]: ntuple name not set!";
+    throw std::runtime_error( error.str() );
+  }
+  if(rhoVar == std::string("")) {
+    std::stringstream error;
+    error << "FieldMapBrBz[ERROR]: rho variable name not set!";
+    throw std::runtime_error( error.str() );
+  }
+  if(zVar == std::string("")) {
+    std::stringstream error;
+    error << "FieldMapBrBz[ERROR]: z variable name not set!";
+    throw std::runtime_error( error.str() );
+  }
+  if(BrhoVar == std::string("")) {
+    std::stringstream error;
+    error << "FieldMapBrBz[ERROR]: Brho variable name not set!";
+    throw std::runtime_error( error.str() );
+  }
+  if(BzVar == std::string("")) {
+    std::stringstream error;
+    error << "FieldMapBrBz[ERROR]: Bz variable name not set!";
+    throw std::runtime_error( error.str() );
+  }
+
+  std::cout << std::endl;
+  std::cout << "Ntuple name:   " << NtupleName << std::endl;
+  std::cout << "rho  Var name: " << rhoVar    << std::endl;
+  std::cout << "z    Var name: " << zVar      << std::endl;
+  std::cout << "Brho Var name: " << BrhoVar   << std::endl;
+  std::cout << "Bz   Var name: " << BzVar     << std::endl;
+  std::cout << std::endl;
+
   TTree *tree;
-  file->GetObject(strs[0].c_str(), tree);
+  file->GetObject(NtupleName.c_str(), tree);
   if (not tree) {
     std::stringstream error;
-    error << "FieldMap[ERROR]: Tree " << strs[0] << " not found in file: " << filename;
+    error << "FieldMapBrBz[ERROR]: Tree " << NtupleName << " not found in file: " << filename;
     throw std::runtime_error( error.str() );
   }
 
   float r, z, Br, Bz;
-  checkBranch( tree->SetBranchAddress(strs[1].c_str(), &r) );
-  checkBranch( tree->SetBranchAddress(strs[2].c_str(), &z) );
-  checkBranch( tree->SetBranchAddress(strs[3].c_str(), &Br) );
-  checkBranch( tree->SetBranchAddress(strs[4].c_str(), &Bz) );
-
-  const int elements = nRho*nZ;
+  checkBranch( tree->SetBranchAddress(rhoVar.c_str(),  &r) );
+  checkBranch( tree->SetBranchAddress(zVar.c_str(),    &z) );
+  checkBranch( tree->SetBranchAddress(BrhoVar.c_str(), &Br));
+  checkBranch( tree->SetBranchAddress(BzVar.c_str(),   &Bz));
+
+  zStep      = -1;
+  rhoStep    = -1;
+  CoorsOrder = std::string("");
   const int treeEntries = tree->GetEntries();
-  if ( elements != treeEntries ) {
+  fieldMap.reserve(treeEntries);
+  for(int i=0;i<treeEntries;i++) {
+    tree->GetEntry(i);
+
+    if(i == 0) {
```

Comment:
I just did this. Now it is possible that a field map can contain one or mare variables ordered from high-to-low. I am testing this right now.